app/graphql/mutations/attach_files_to_sample.rb

```diff
@@ -6,16 +6,24 @@ class AttachFilesToSample < BaseMutation
     description 'Attaches files to sample.'
     argument :files, [String], required: true, description: 'A list of files (signedBlobId) to attach to the sample'
     argument :sample_id, ID,
-             required: true,
-             description: 'The Node ID of the sample to be updated.'
+             required: false,
+             description: 'The Node ID of the sample to be updated. For example, `gid://irida/Sample/a84cd757-dedb-4c64-8b01-097020163077`' # rubocop:disable Layout/LineLength
+    argument :sample_puid, ID, # rubocop:disable GraphQL/ExtractInputType
+             required: false,
+             description: 'Persistent Unique Identifier of the sample. For example, `INXT_SAM_AAAAAAAAAA`.'
+    validates required: { one_of: %i[sample_id sample_puid] }
 
     field :errors, GraphQL::Types::JSON, null: false, description: 'Errors that prevented the mutation.'
     field :sample, Types::SampleType, null: false, description: 'The updated sample.'
     field :status, GraphQL::Types::JSON, null: false, description: 'The status of the mutation.'
 
-    def resolve(sample_id:, files:)
-      sample = IridaSchema.object_from_id(sample_id, { expected_type: Sample })
-      files_attached = Attachments::CreateService.new(current_user, sample, { files: }).execute
+    def resolve(args)
+      sample = if args[:sample_id]
+                 IridaSchema.object_from_id(args[:sample_id], { expected_type: Sample })
+               else
+                 Sample.find(args[:sample_puid])
```

Comment:
This should be a `find_by` instead of `find`. Currently it's trying to find a sample with an `id` of the `sample_puid`

Reply:
Fixed in c362e774 good catch!